kinto/core/storage/postgresql/__init__.py

```diff
@@ -737,14 +739,28 @@ def _format_conditions(self, filters, id_field, modified_field,
 
             if is_like_query:
                 # Operand should be a string.
-                # Add implicit start/end wildchars if none is specified.
+                # Add implicit start/end wildcards if none is specified.
                 if '*' not in value:
                     value = '*{}*'.format(value)
                 value = value.replace('*', '%')
 
             if filtr.operator == COMPARISON.HAS:
                 operator = 'IS NOT NULL' if filtr.value else 'IS NULL'
                 cond = '{} {}'.format(sql_field, operator)
+
+            elif filtr.operator == COMPARISON.CONTAINS_ANY:
+                value_holder = '{}_value_{}'.format(prefix, i)
+                holders[value_holder] = value
+                # Postgres's && operator doesn't support jsonbs.
+                # However, it does support Postgres arrays of any
+                # type. Assume that the referenced field is a JSON
+                # array and convert it to a Postgres array.
+                # N.B. This won't work on JSON objects and will never match.
```

Comment:
No, that's not right.

```
http --auth 'user:user' PUT 'http://localhost:8888/v1/buckets/abc/collections/def/records/r1' 'data:={"user": {"first": "Ethan", "id": 1234}, "aliases": [{"ll": "ls -l"}, {"gti": "git"}, {"vi": "vim"}]}'
http --auth 'user:user' 'http://localhost:8888/v1/buckets/abc/collections/def/records?overlaps_aliases=[{"ll":"ls -l"}]'   # matches
http --auth 'user:user' 'http://localhost:8888/v1/buckets/abc/collections/def/records?overlaps_aliases=[{"sl":"ls -l"}]'   #doesn't match
http --auth 'user:user' 'http://localhost:8888/v1/buckets/abc/collections/def/records?overlaps_user={"id":1234}'   # 503
```

(This uses `overlaps` here because `contains_any` has a different bug...)

Compare with `contains`:

```
http --auth 'user:user' 'http://localhost:8888/v1/buckets/abc/collections/def/records?contains_aliases=[{"ll":"ls -l"}]'   # matches
http --auth 'user:user' 'http://localhost:8888/v1/buckets/abc/collections/def/records?contains_aliases=[{"sl":"ls -l"}]'   #doesn't match
http --auth 'user:user' 'http://localhost:8888/v1/buckets/abc/collections/def/records?contains_user={"id":1234}'  # matches
```

N.B. these behaviors may not match in the memory backend.